### DISCUSSION TITLE
Upgrade guide (0.42.0 to 0.45.3)

### DISCUSSION THREAD

**Discussion by davetapley** [fibo/trunx#60] [General]
Just trying to get caught up and it looks like there's be a **lot** of refactoring in https://github.com/fibo/trunx/compare/v0.42.0...v0.45.2.

Are there any notes / guidance anywhere? 🤔

**Comment by fibo** [comment]
hi @davetapley 

yes there were few improvements.

The components did not changed that much: you should be able to use TS types and inline docs to adapt to new changes.

The main new feature are the `bulma` and `classNames` helpers: they provide an interface similar to `classnames` package (that was a trunx dependency until 0.42) but with TS autocompletion.

So for example, if you want to create a footer (now there is no Footer component by the way) you can do something like this



<img width="933" alt="Screenshot 2023-08-20 at 17 37 11" src="https://github.com/fibo/trunx/assets/176730/27a8f2c6-0d79-4f7d-a9ae-4067367ee365">

and you will get all bulma classes autocompleted.

The `classNames` helper let you add your own classes, you can do something like

```ts
import {
 classNames,
 BulmaClassName,
 ClassNamesArg
} from "trunx"

// Add other classes.
type MyClassName = BulmaClassName | "Foo" | "Bar"

export const myClassNames = (...args: ClassNamesArg<MyClassName>[]) => classNames(...args)

```


You can find also some snippets in trunx website (still not completed, would need some help to finish or more time) here: https://github.com/fibo/trunx/tree/main/webapps/trunx-docs/app

I hope this answer your question, if you mean guidance in terms of a roadmap or a place where we discuss this things, I am open to proposals and feedbacks.

Currently I am using trunx for few side projects, so I am going to keep maintain it and improving it.

But in general it is almost done and the interface will not change that much, I mean, with this new `bulma` and `classNames` CSS helpers I think the package provides a lot of flexibility and I am enjoining a lot using it.

The cons are that I do not have so much time to finish documenting it.
